### PR TITLE
TRAD-210 swap with route specification

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: proto/api.proto [hunk 1]
```diff
@@ -288,6 +288,17 @@ service Api {
     };
   }
 
+  rpc PostMultiRouteTradeSwap(MultiRouteTradeSwapRequest) returns (MultiRouteTradeSwapResponse) {
```
Comment (aspin) [inline proto/api.proto]: naming: I don't think `Multi` should be part of this name. This is a swap with a route specification, and it's valid if the user decides to give a route specification of size 1. So I think `PostRouteTradeSwap` or `PostRouteSwap` would be better

File: proto/api.proto [hunk 2]
```diff
@@ -733,13 +744,31 @@ message ProjectQuote {
...
 
-message TradeSwapRequest{
+message TradeSwapRequest {
   Project project = 1;
   string owner = 2;
   string inToken = 3;
-  string outToken = 4;
-  double inAmount = 5;
-  double slippage = 6;
+  string middleToken = 4;
```
Comment (aspin) [inline proto/api.proto]: does middle token need to be made to the original request?

File: proto/api.proto (hunk 1 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: ok done.

File: proto/api.proto (hunk 2 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: no, removed it

**Review by aspin — APPROVED** [review]

File: proto/api.proto [hunk 3]
```diff
@@ -289,6 +289,17 @@ service Api {
     };
   }
 
+  rpc PostRouteTradeSwap(RouteTradeSwapRequest) returns (RouteTradeSwapResponse) {
+    option(google.api.http) = {
+      get: "/api/v1/amm/route-trade-swap"
```
Comment (aspin) [inline proto/api.proto]: endpoint needs to be updated: should be `/api/v1/trade/route-swap`

File: proto/api.proto [hunk 4]
```diff
@@ -289,6 +289,17 @@ service Api {
...
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      description: "routes swap(s) from supported amms";
+      summary: "Routes swaps";
+      tags: "AMMs";
```
Comment (aspin) [inline proto/api.proto]: `"Trade/AMM"`

File: proto/api.proto [hunk 5]
```diff
@@ -289,6 +289,17 @@ service Api {
     };
   }
 
+  rpc PostRouteTradeSwap(RouteTradeSwapRequest) returns (RouteTradeSwapResponse) {
+    option(google.api.http) = {
+      get: "/api/v1/amm/route-trade-swap"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      description: "routes swap(s) from supported amms";
```
Comment (aspin) [inline proto/api.proto]: description for consistency: "Generates an unsigned transaction object for the AMM swap on the specified route"

summary for consistency: "Unsigned AMM swap transaction along route"

**Review by aspin — CHANGES_REQUESTED** [review]
see changes in https://github.com/bloXroute-Labs/solana-trader-proto/pull/29/files#diff-ecf0878b123623e280a40f3e3d75af4267f11cfe7138753e671654c0ba05f228R264

File: proto/api.proto (hunk 3 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

File: proto/api.proto (hunk 4 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

File: proto/api.proto (hunk 5 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

File: proto/api.proto [hunk 6]
```diff
@@ -311,6 +311,17 @@ service Api {
...
   }
 
+  rpc PostRouteTradeSwap(RouteTradeSwapRequest) returns (RouteTradeSwapResponse) {
+    option(google.api.http) = {
+      get: "/api/v1/amm/route-swap"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      description: "Generates an unsigned transaction object for the AMM swap on the specified route";
+      summary: "Unsigned AMM swap transaction along route";
+      tags: "Trade/AMM";
```
Comment (aspin) [inline proto/api.proto]: actually should be `["Trade", "AMM"]` now, after some review changes on my end

File: proto/api.proto [hunk 7]
```diff
@@ -311,6 +311,17 @@ service Api {
     };
   }
 
+  rpc PostRouteTradeSwap(RouteTradeSwapRequest) returns (RouteTradeSwapResponse) {
+    option(google.api.http) = {
+      get: "/api/v1/amm/route-swap"
```
Comment (aspin) [inline proto/api.proto]: endpoint needs to be `/api/v1/trade/route-swap`

**Review by aspin — CHANGES_REQUESTED** [review]

File: proto/api.proto (hunk 7 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

File: proto/api.proto (hunk 6 above)
Comment (aspin) [reply proto/api.proto]: bump

**Comment by aspin** [comment]
bump on the tags comment

File: proto/api.proto (hunk 6 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

**Review by aspin — APPROVED** [review]

**Pull request merged**